lib/url.js

```diff
@@ -363,10 +376,21 @@ function urlFormat(obj) {
 }
 
 Url.prototype.format = function() {
-  var auth = this.auth || '';
-  if (auth) {
-    auth = encodeURIComponent(auth);
+  var auth = '';
+
+  if (this.username) {
+    auth = encodeURIComponent(this.username);
+
+    // If we had a blank password such as `user:` for our auth part, then we want to reconstruct it correctly.
+    if (this.password !== null) {
+      auth += ':' + encodeURIComponent(this.password);
+    }
+  } else if (this.auth) {
```

Comment:
I think this else if can be removed, because I don't see how someone would create an object which is an instance of the `Url` module and has a value for `auth` but not `username`.